census.go

```diff
@@ -432,6 +434,147 @@ func (v *vocdoniHandler) censusQueueInfo(msg *apirest.APIdata, ctx *httprouter.H
 	return ctx.Send(data, http.StatusOK)
 }
 
+type Token struct {
+	Address    string `json:"address"`
+	Blockchain string `json:"blockchain"`
+}
+
+func (v *vocdoniHandler) censusTokenNFT(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	req := struct {
+		Tokens []*Token `json:"tokens"`
+	}{}
+	if err := json.Unmarshal(msg.Data, &req); err != nil {
+		return err
+	}
+
+	// check valid tokens length
+	if len(req.Tokens) > 3 || len(req.Tokens) == 0 {
+		return fmt.Errorf("invalid number of NFT tokens, bounds between 1 and 3")
+	}
+	// check valid tokens
+	for _, token := range req.Tokens {
+		if _, err := v.airstack.BlockchainToTokenBlockchain(token.Blockchain); err != nil {
+			return fmt.Errorf("invalid blockchain for token %s provided", token.Address)
+		}
+	}
+	return v.censusToken(req.Tokens, ctx)
+}
+
+func (v *vocdoniHandler) censusTokenERC20(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	req := struct {
+		Token Token `json:"token"`
+	}{}
+	if err := json.Unmarshal(msg.Data, &req); err != nil {
+		return err
+	}
+
+	// check valid token
+	if _, err := v.airstack.BlockchainToTokenBlockchain(req.Token.Blockchain); err != nil {
+		return fmt.Errorf("invalid blockchain for token %s provided", req.Token.Address)
+	}
+	tks := make([]*Token, 1)
+	tks[0] = &req.Token
+	return v.censusToken(tks, ctx)
+}
+
+func (v *vocdoniHandler) censusToken(tokens []*Token, ctx *httprouter.HTTPContext) error {
```

Comment:
We are repeating a lot of code here. All the census functions should be merged into a single one, but let's do it in a future refactor.